File: Client/src/app/models/user/user-dto.ts
export class UserDto {
  id: string;
  username: string;
  email: string;
  firstName?: string;
  lastName?: string;
  phone?: string;
  createdAt: Date;
  isActive: boolean;
  followersCount: number;
  followingCount: number;

  constructor(
    id: string,
    username: string,
    email: string,
    createdAt: Date,
    isActive: boolean,
    followersCount: number,
    followingCount: number,
    firstName?: string,
    lastName?: string,
    phone?: string,
  ) {
    this.id = id;
    this.username = username;
    this.email = email;
    this.firstName = firstName;
    this.lastName = lastName;
    this.phone = phone;
    this.createdAt = createdAt;
    this.isActive = isActive;
    this.followersCount = followersCount;
    this.followingCount = followingCount;
  }
}

File: Shared/src/Models/User/UserDTO.cs
using System.ComponentModel.DataAnnotations;



public  class UserDTO
{

     public Guid Id { get; set; }

    [Required]
    public required string Username { get; set; }

    [Required]
    public required string Password { get; set; }

    [Required]
    public required string Email { get; set; }

    [Required]
    public Guid StreamId { get; set; }

    public string? FirstName { get; set; }

    public string? LastName { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public bool IsLive { get; set; } = false;

    public ICollection<Guid> FollowerIds{ get; set; } = new List<Guid>();

    public ICollection<Guid> FollowingIds { get; set; } = new List<Guid>();

   
}